### ISSUE TITLE
[Blazor] Mysterious phenomenon of using Math.Truncate in Blazor

### ISSUE THREAD

**Issue by mkArtakMSFT** [mono/mono#13428]
_From @txgz999 on Thursday, 24 January 2019 01:52:33_

### Describe the bug
When I use Math.Truncate in the definition of a property used in data binding, a print window pops up

### To Reproduce
Create a Blazor project in VS2017, add a page with the following content:

```
@page "/mathtruncate"

@test

@functions {
    private int test
    {
        get
        {
            // var k = Math.Max(1, 2); // works fine
            var k = Math.Truncate(20d); // a print window pops up
            return 1;
        }
    }
}
```

### Expected behavior
Browse to that page, you would see just content, which is value 1

### Screenshots
![mathtruncate](https://user-images.githubusercontent.com/29514515/51648932-8e7e5700-1f50-11e9-90b9-00c04ce0ed20.png)

### Additional context
Other Math methods such as Math.Max work as expected.

_Copied from original issue: aspnet/AspNetCore#6973_

**Comment by mkArtakMSFT** [comment]
_From @rynowak on Thursday, 24 January 2019 18:51:40_

Wow! This is an amazing bug. Can you file this at https://github.com/mono/mono - this is probably an issue in the underlying runtime.

**Comment by mkArtakMSFT** [comment]
_From @txgz999 on Friday, 25 January 2019 14:19:35_

 Thanks a lot, Ryan, for verifying the issue. I agree to your speculation. Regarding your suggestion, I am not sure if i have enough knowledge to explain the issue well. If you like, please go ahead to create a ticket there.


    On Thursday, January 24, 2019, 1:52:20 PM EST, Ryan Nowak <notifications@github.com> wrote:  
 
 
Wow! This is an amazing bug. Can you file this at https://github.com/mono/mono - this is probably an issue in the underlying runtime.

**Comment by mkArtakMSFT** [comment]
_From @gulbanana on Friday, 01 March 2019 08:28:31_

I've experienced this too, it's very amusing. I have a repro but it's a lot more complex than just calling Math.Truncate (which we don't even use!). Just, doing a lot of work in WASM leads to the print dialog opening..

**Comment by kjpou1** [comment]
Hey @mkArtakMSFT 

We will need more information.  I have created a straight wasm example here:  https://kjpou1.github.io/WasmMathTrunc/

This seems to be working as expected with no popup.

``` csharp
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Net.Http;
using WebAssembly;

namespace WasmMathTrunc
{
    public class Program
    {
        public static void Main()
        {
            Console.WriteLine("Hello from MathTruncate");
        }
        private int test ()
        {
            Console.WriteLine("Before MathTrunc");
            // var k = Math.Max(1, 2); // works fine
            var k = Math.Truncate(20d); // a print window pops up
            Console.WriteLine($"After MathTrunc {k}");
            return 1;
        }        
    }

}
```

The console prints the following:

```
>>mono_wasm_runtime_ready
mono.js:1 ﻿Hello from MathTruncate
mono.js:1 Before MathTrunc
mono.js:1 After MathTrunc 20
(index):87 Total Memory Exit WASM: 67108864
```

Have tested on Edge and Chrome on windows.
FF, Chrome, Safari on Mac

None of the above present a popup.

**Comment by kjpou1** [comment]
Also, just a thought.  Somewhere in the javascript code on the Blazor side there is a call to `print` instead of to the Module's `print` function.

**Comment by kg** [comment]
If you modify the JS for your reproduction case to replace window.print and set a breakpoint there, that should give you a stack trace that will point us directly to the bug. Something like (off the top of my head, might not compile):

```javascript
window.print = function () {
  console.logError("window.print called");
};
```

**Comment by kjpou1** [comment]
@txgz999 can you have a look above

**Comment by txgz999** [comment]
The bug I reported is regarding Blazor 0.7 running in VS2017. I tried kg's suggestion last week and I did hit the breakpoint there. Unfortunately I did not have time to analyze the stack trace at that time.

Now I have Blazor 0.9 installed running in VS2019 RC3. I just checked this bug disappeared. Thanks you all for your help.  @gulbanana, do you still experience this issue with Blazor 0.9?

**Comment by kjpou1** [comment]
Thanks for the info @txgz999.  Will wait a little longer for response before resolving this issue.

**Comment by kjpou1** [comment]
Am closing this as original poster confirms that upgrade to Blazor 0.9 fixes the issue.  If this is not the case please open a new issue.

**Comment by lewing** [comment]
This was likely fixed in https://github.com/mono/mono/pull/11602 but shipped in blazor 0.7
